modules/schema/src/main/java/org/apache/ignite/internal/schema/marshaller/reflection/Marshaller.java

```diff
@@ -215,6 +232,10 @@ public Object readObject(Row reader) throws MarshallerException {
         @Override
         public void writeObject(Object obj, RowAssembler writer)
                 throws MarshallerException {
+            if (readOnly) {
```

Comment:
Why do we need such a limitation? To me the previous behaviour is much better, because you got here an exception describing a root cause of the issue (setting null to a non-null field). Instead we now got here an abstract "Can't assemple row by object".

BTW there is a typo in "assemble".

Reply:
Furthermore the current behaviour will be changed in case the user provide a null value supplier explicitly.

Reply:
Read only logic removed.